share/spack/setup-env.sh

```diff
@@ -237,11 +237,12 @@ fi;
 # Set up modules and dotkit search paths in the user environment
 #
 
-read -r -d 'EOCs' _python_command << EOCs
+_python_command=$(cat << EOCs
```

Comment:
Huh?
```bash
￼  _python_command='line1\nline2\nline3'
echo -e "${￼  _python_command}"
```

Reply:
@junghans I can do that, and it would also be more portable. The fact is I was looking for a way to maintain the commands on different lines (and avoid a line that finishes 3 screens away). 

I still need to test this with different shells. If there are some for which this change breaks things, I'll probably do like you suggest. @tgamblin Would you agree with this single, super-unreadable line in case?

Reply:
Looking at http://pubs.opengroup.org/onlinepubs/9699919799/utilities/echo.html, the most portable way would be using the sh builtin `printf`: 
```
printf  "%s\n%s\n%s" "line1" "line2" "line3"
```
That could even include the ` tr -d ' '` part and then:
```
_assignment_command=$(spack-python -c "exec('$(printf bla)')")
```

Reply:
Ok: apparently I broke `zsh` support with this. `dash` was broken before and remain such. Are there other shells I am not aware of and we are interested in?

@junghans Thanks for the hints. Do you know if there's a way to obtain the same string as with `printf`, but write it on multiple lines within the file?  Writing this single line command with a sort of literal (which is what I need for `spack-python -c`) I feel like I am punching readability directly in the face.